### DISCUSSION TITLE
Add option to control environment variables from a .env file and option to add labels for Traefik

### DISCUSSION THREAD

**Discussion by theriaultjulien** [louislam/dockge#16] [Feature Request]
It would be nice to have the option to choose if we want to specify Environment Variables explicitly or use a .env file and we just need to write the path of the .env file to use.

Also 2nd suggestion would be to add a way to add labels for us Traefik users.

**Comment by anthonysmillie** [comment]
With you on the env_file parameter.

Even nicer would be the ability to edit said env from the interface if an env_file has been defined for a certain container.

**Comment by ToshY** [comment]
> It would be nice to have the option to choose if we want to specify Environment Variables explicitly or use a .env file and we just need to write the path of the .env file to use.

Maybe I don't understand this (and because I'm curious), could you explain why there is a need for "specify Environment Variables explicitly or use a .env file" when compose already handles [environment variables precedence](https://docs.docker.com/compose/environment-variables/envvars-precedence/)?

> Also 2nd suggestion would be to add a way to add labels for us Traefik users.

The `labels` would be a nice to have (also for non-Traefik users).

**Reply by theriaultjulien** [reply]
I'm not sure what you're inferring with the environment variables precedence, but it's not linked to my suggestion. If I can explain my suggestion another way, I would just want to have in the UI a dropdown to choose between setting environment variables manually by writing them in the textboxes (like it is right now), or to simply specify the path of a .env file (that I would create and manually place where I want. Or like @anthonysmillie said in previous comment, even better if we could edit the .env file that we specify directly in the UI). 

I just like to use .env files better for some containers that need private information in environment variables, since I'm pushing all my compose files to my github repository. I don't want private information to be saved on github in cleartext. This way my private info stays in .env files on my server and my compose files are clean.

Hope this clears it up for you.

**Comment by RichyHBM** [comment]
If we could provide a custom template for the stack, we could add any label we needed, then just have dockage fill in the template with things like the stack name, service name, service image...

**Comment by trythatagain** [comment]
I've just found this project and like what I see, I'm using OMV (Open media vault) and use the Compose plugin, and 1 feature I really like is the  'global' env file, where I store appdata, tz etc ... This means just setting once a load of common environment  variables, and just use the '$var' name in my compose file.
Plus if this could be on the start page say next to the 'Home', 'Console' etc would be nice.

**Comment by jtenniswood** [comment]
I would love to see the .env file editable in the UI, handy for things like immich that relies on it.

**Comment by davidh2k** [comment]
Gonna add that for Managing a docker installation with Dockge that is using traefik you currently need to manually edit the docker-compose.yaml for adding the labels and also accessing the server via ssh to place/edit the per stack .env file. There are however different approaches as to how people might organize their servers (complete stack for the whole server vs. one stack for every application).

Example part of compose: (This could also be nice if you could copy this from some kind of template, since it stays the same for most containers (exceptions do exists though))

```
    labels:
      - traefik.enable=true
      - traefik.http.routers.${ServiceName}.entrypoints=websecure
      - traefik.http.routers.${ServiceName}.rule=Host(`${ServiceName}.${Domain}`)
      - traefik.http.routers.${ServiceName}.tls=true
      - traefik.http.routers.${ServiceName}.tls.certresolver=le
      - traefik.http.services.${ServiceName}.loadbalancer.server.port=80
```

Example .env:

```
ServiceName=audiobookshelf
Domain=not.my-domain.org

```
